x/gamm/twap/README.md

```diff
@@ -1,9 +1,102 @@
+
+## Code layout
+
+**api.go** is the main file you should look at as a user of this module.
+
+**logic.go** is the main file you should look at for how the TWAP implementation works.
+
+- types/* - Implement TwapRecord, GenesisState. Define AMM interface, and methods to format keys.
+- api.go - Public API, that other users / modules can/should depend on
+- hook_listener.go - Defines hooks & calls to logic.go, for triggering actions on 
+- keeper.go - generic SDK boilerplate (defining a wrapper for store keys + params)
+- logic.go - Implements all TWAP module 'logic'. (Arithmetic, defining what to get/set where, etc.)
+- module.go - SDK AppModule interface implementation.
+- store.go - Managing logic for getting and setting things to underlying stores
+
+## Store layout
+
+Every pool has a TWAP stored in state for every asset pair.
+
+## Testing Methodology
+
+The pre-release testing methodology planned for the twap module is:
+
+- [ ] Using table driven unit tests to test all foreseen cases the module can be within
+    - hook testing
+        - All swaps correctly trigger twap record updates
+        - Create pools cause records to be created
+    - store
+        - EndBlock triggers all relevant twaps to be saved correctly
+        - Block commit wipes temporary stores
+    - logic
+        - Make tables of expected input / output cases for:
+          - getMostRecentRecord
+          - getInterpolatedRecord
+          - updateRecord
+          - computeArithmeticTwap
+        - Test overflow handling in all relevant arithmetic
+        - Complete testing code coverage (up to return err lines) for logic.go file
+    - API
+        - Unit tests for the public API, under foreseeable setup conditions
+- [ ] Integration into the Osmosis simulator
+    - The osmosis simulator, simulates building up complex state machine states, in random ways not seen before. We plan on, in a property check, maintaining expected TWAPs for short time ranges, and seeing that the keeper query will return the same value as what we get off of the raw price history for short history intervals.
+    - Not currently in scope for release blocking, but planned: Integration for gas tracking, to ensure gas of reads/writes does not grow with time.
+- [ ] Mutation testing usage
+    - integration of the TWAP module into go mutation testing: https://github.com/osmosis-labs/go-mutesting
+        - The success we've seen with the tokenfactory module, is it succeeds at surfacing behavior for untested behavior.
+          e.g. if you delete a line, or change the direction of a conditional, does a test catch it.
+    - We expect to get this to a state, where after mutation testing is ran, the only items it mutates, that is not caught in a test, is: Deleting `return err`, or `panic` lines, in the situation where that error return or panic isn't reachable.
```

Comment:
- While this is awesome, do folks need to know such detail about mutation testing?
- Suggest paraphrasing to make it clearer. A lot of context for one sentence